app/routes.tsx

```diff
@@ -122,6 +112,18 @@ export function Routes() {
   const store = useStore();
   const diagnosticPermission = useHasUserGivenDiagnosticPermissions();
 
+  useEffect(() => {
+    if (process.env.SENTRY_DSN && diagnosticPermission) {
+      Sentry.init({
+        dsn: process.env.SENTRY_DSN,
+        beforeSend(event) {
+          if (!diagnosticsEnabled) return null;
+          return event;
+        },
+      });
+    }
+  }, []);
```

Comment:
missing dependency ?

Reply:
no, we want this to run only once (componentDidMount).